src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -10302,6 +10302,24 @@ instruct array_equalsC(iRegP_R11 ary1, iRegP_R12 ary2, iRegI_R10 result,
   ins_pipe(pipe_class_memory);
 %}
 
+// fast ArraysSupport.vectorizedHashCode
+instruct arrays_hashcode(iRegP_R11 ary, iRegI_R12 cnt, iRegI_R10 result, immI basic_type,
+                         iRegLNoSp tmp1, iRegINoSp tmp2,
+                         iRegINoSp tmp3, iRegLNoSp tmp4, rFlagsReg cr)
+%{
+  match(Set result (VectorizedHashCode (Binary ary cnt) (Binary result basic_type)));
+  effect(TEMP tmp1, TEMP tmp2, TEMP tmp3, TEMP tmp4,
+         USE_KILL ary, USE_KILL cnt, USE basic_type, KILL cr);
```

Comment:
I don't think we need to add `USE basic_type` to the effect list. It is an immediate input which is already there in the match rule.